pandas/io/stata.py

```diff
@@ -1931,7 +1937,9 @@ def read_stata(
 
 
 def _open_file_binary_write(
-    fname: FilePathOrBuffer, compression: Union[str, Mapping[str, str], None],
+    fname: FilePathOrBuffer,
+    compression: Union[str, Mapping[str, str], None],
```

Comment:
can also add Compression to typing.py

Reply:
This wasn't me, but the styling; compression as a kwarg has different types in different places (most often str, but also `Optional[Union[str, Mapping[str, Any]]]`, which is similar), so I don't think there's a useful common type to be made.

Reply:
hmm interesting, we ought to fix that. can you create an issue about this?

Reply:
I'm not certain it's wrong - sometimes the args is required, but could be None, other times it's not required, but the default value can be None or str (e.g., "infer").